src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -145,36 +145,36 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
 
         public static void ToDeserializeCall(this CodeWriter writer, JsonSerialization serialization, CodeWriterDelegate element, ref string destination)
         {
-            destination = writer.GetTemporaryVariable(destination);
+            var destinationDeclaration = new CodeWriterDeclaration(destination);
 
             if (serialization is JsonValueSerialization valueSerialization)
             {
-                writer.Append($"var {destination:D} =")
+                writer.Append($"var {destinationDeclaration:D} =")
```

Comment:
First time a declaration is used with `:D` suffix it gets an `ActualName` assigned that is unique in this scope.